qt/python/mantidqt/mantidqt/widgets/codeeditor/test/test_multifileinterpreter.py

```diff
@@ -49,6 +51,24 @@ def test_open_file_in_new_tab_no_import_added(self):
                 widget.open_file_in_new_tab(test_string)
         self.assertNotIn(MANTID_API_IMPORT, widget.current_editor().editor.text())
 
+    def test_save_as_closes_the_original_tab(self):
+        test_string = "Test file\n"
+        widget = MultiPythonFileInterpreter()
+        mock_open_func = mock.mock_open(read_data=test_string)
+        with mock.patch(widget.__module__ + ".open", mock_open_func, create=True):
+            with mock.patch(PERMISSION_BOX_FUNC, lambda: True):
+                with tempfile.TemporaryDirectory() as temp_dir:
+                    with mock.patch(
+                        "mantidqt.widgets.codeeditor.interpreter.EditorIO.ask_for_filename",
+                        lambda s: os.path.join(temp_dir, "test_save_as_leaves_original_marked_unmodified"),
+                    ):
+                        widget.open_file_in_new_tab(test_string)
+                        widget.save_current_file_as()
+        # There's one tab when you create the editor, then another one was added when opening the mock file. Previously
+        # clicking 'Save Script as' would leave the original tab open, so you'd end up with 3 tabs instead of 2
```

Comment:
Instead of referencing what used to happen, I think it'd be better to write something like `Clicking Save Script as should close the original tab`, something to that effect anyway.